bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/installcomponents/desrializer/ComponentsDesrializer.java

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * Components JSON response desrializer
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class ComponentsDesrializer implements JsonDeserializer<List<ComponentVO>>
+{
+
+	@Override
+	public List<ComponentVO> deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+			throws JsonParseException
+	{
+		List<ComponentVO> componentVOs = new ArrayList<>();
+		JsonArray jsonArray = json.getAsJsonArray();
+		for (int i = 0; i < jsonArray.size(); i++)
+		{
+			JsonElement jsonElement = jsonArray.get(i);
+
+			JsonObject mainJsonObject = jsonElement.getAsJsonObject();
+			ComponentVO componentVO = new ComponentVO();
+			String createdAt = mainJsonObject.get("created_at") == null || mainJsonObject.get("created_at").isJsonNull() //$NON-NLS-1$ //$NON-NLS-2$
+					? null
+					: mainJsonObject.get("created_at").getAsString(); //$NON-NLS-1$
+			boolean featured = mainJsonObject.get("featured") == null || mainJsonObject.get("featured").isJsonNull() //$NON-NLS-1$ //$NON-NLS-2$
+					? false
+					: mainJsonObject.get("featured").getAsBoolean(); //$NON-NLS-1$
+			String name = mainJsonObject.get("name") == null || mainJsonObject.get("name").isJsonNull() ? null //$NON-NLS-1$ //$NON-NLS-2$
+					: mainJsonObject.get("name").getAsString(); //$NON-NLS-1$
+			String namespace = mainJsonObject.get("namespace") == null || mainJsonObject.get("namespace").isJsonNull() //$NON-NLS-1$ //$NON-NLS-2$
+					? null
+					: mainJsonObject.get("namespace").getAsString(); //$NON-NLS-1$
+			componentVO.setCreatedAt(createdAt);
+			componentVO.setFeatured(featured);
+			componentVO.setName(name);
+			componentVO.setNamespace(namespace);
+
+			if (mainJsonObject.get("latest_version").isJsonNull()) //$NON-NLS-1$
+			{
+				componentVOs.add(componentVO);
+				continue;
+			}
+			JsonObject detailsJsonObject = mainJsonObject.getAsJsonObject("latest_version"); //$NON-NLS-1$
+
+			String componentHash = detailsJsonObject.get("component_hash") == null //$NON-NLS-1$
+					|| detailsJsonObject.get("component_hash").isJsonNull() ? null //$NON-NLS-1$
+							: detailsJsonObject.get("component_hash").getAsString(); //$NON-NLS-1$
+			String details = detailsJsonObject.get("description") == null //$NON-NLS-1$
+					|| detailsJsonObject.get("description").isJsonNull() ? null //$NON-NLS-1$
+							: detailsJsonObject.get("description").getAsString(); //$NON-NLS-1$
+			createdAt = detailsJsonObject.get("created_at") == null || detailsJsonObject.get("created_at").isJsonNull() //$NON-NLS-1$ //$NON-NLS-2$
+					? null
+					: detailsJsonObject.get("created_at").getAsString(); //$NON-NLS-1$
+			String url = detailsJsonObject.get("url") == null || detailsJsonObject.get("url").isJsonNull() ? null //$NON-NLS-1$ //$NON-NLS-2$
+					: detailsJsonObject.get("url").getAsString(); //$NON-NLS-1$
+			String version = detailsJsonObject.get("version") == null || detailsJsonObject.get("version").isJsonNull() //$NON-NLS-1$ //$NON-NLS-2$
+					? null
```

Comment:
For the json keys we could create local constants and use them multiple times. For example, I see `created_at` used 6 times

Reply:
Done